src/components/ListInput.tsx

```diff
@@ -0,0 +1,44 @@
+import styled from 'styled-components'
+import useList from '../hooks/useList'
+import Button from './Button'
+
+interface Props {
+  items: string[]
+  onChange: (items: string[]) => void
+}
+
+export default function ListInput(props: Props) {
+  const { value, handleChange, handleSave, handleReset, isDirty } = useList(
+    props
+  )
+
+  return (
+    <Wrapper>
+      <Textarea rows={20} value={value} onChange={handleChange} />
+      <Button disabled={!isDirty} onClick={handleReset}>
+        cancel
+      </Button>
+      <Button disabled={!isDirty} onClick={handleSave}>
+        save
+      </Button>
+    </Wrapper>
+  )
+}
+
+const Textarea = styled.textarea`
+  font-size: 1em;
+  font-family: inherit;
+  padding: 1px;
+  border: 1px solid white;
+  line-height: 1.1em;
+  color: var(--color-blue);
+`
+
+const Wrapper = styled.section`
+  display: grid;
+  grid-template-columns: repeat(2, min-content);
+
+  ${Textarea} {
+    grid-column: span 2;
```

Comment:
Why don't you give your `Textarea` this style?

Reply:
🤔 🤷 I think I didn't want to bleed the layout into the child component, but it doesn't really matter here. I think I will follow your suggestion.